library/core/src/str/iter.rs

```diff
@@ -656,7 +656,7 @@ impl<'a, P: Pattern> SplitInternal<'a, P> {
         None
     }
 
-    #[inline]
+    #[inline(always)]
```

Comment:
maybe we should back out the `inline(always)` and trust LLVM to make the right decision once we've emitted the `inline` hint at all?

Reply:
hm.